File: 1333. Filter Restaurants by Vegan-Friendly, Price and Distance/_1333.py
class Solution:
    def filterRestaurants(self, restaurants, veganFriendly: int, maxPrice: int, maxDistance: int):
        from operator import itemgetter
        res = list()
        for i in restaurants:
            if veganFriendly:
                if i[2] and i[3]<=maxPrice and i[4]<=maxDistance:
                    res.append([i[0], i[1]])
            else:
                if i[3]<=maxPrice and i[4]<=maxDistance:
                    res.append([i[0], i[1]])

        res.sort(key=itemgetter(1,0), reverse=True)

        return [x[0] for x in res]


a = Solution()
print(a.filterRestaurants(restaurants = [[1,4,1,40,10],[2,8,0,50,5],[3,8,1,30,4],[4,10,0,10,3],[5,1,1,15,1]], veganFriendly = 0, maxPrice = 30, maxDistance = 3))

File: 1333. Filter Restaurants by Vegan-Friendly, Price and Distance/1333.go
package leetcode

import (
	"sort"
)

type IntSlice [][]int

func (s IntSlice) Len() int { return len(s) }
func (s IntSlice) Swap(i, j int){ s[i], s[j] = s[j], s[i] }
func (s IntSlice) Less(i, j int) bool {
	if s[i][1] < s[j][1]{
		return false
	}else if s[i][1] > s[j][1]{
		return true
	}else{
		if s[i][0] < s[j][0]{
			return false
		}else {
			return true
		}
	} 
}


func filterRestaurants(restaurants [][]int, veganFriendly int, maxPrice int, maxDistance int) []int {
	res := make([][]int, 0)
	
	for _,v := range restaurants{
		if veganFriendly==1{
			if v[2] == 1 && v[3]<=maxPrice && v[4]<=maxDistance{
				tmp := []int{v[0], v[1]}
				res = append(res, tmp)
			}
		}else{
			if v[3]<=maxPrice && v[4]<=maxDistance{
				tmp := []int{v[0], v[1]}
				res = append(res, tmp)
			}
		}
	}

	sort.Sort(IntSlice(res))

	l := make([]int, 0, len(res))
	for _, v := range res{
		l = append(l, v[0])
	}

	return l
}